sdks/python/apache_beam/runners/portability/flink_runner.py

```diff
@@ -32,26 +33,39 @@
 
 class FlinkRunner(portable_runner.PortableRunner):
   def default_job_server(self, options):
-    flink_master = options.view_as(FlinkRunnerOptions).flink_master
-    if flink_master == '[local]' or sys.version_info < (3, 6):
+    flink_master = self.add_http_scheme(
+        options.view_as(FlinkRunnerOptions).flink_master)
+    options.view_as(FlinkRunnerOptions).flink_master = flink_master
+    if flink_master in ['[local]', '[auto]'] or sys.version_info < (3, 6):
       portable_options = options.view_as(pipeline_options.PortableOptions)
       if flink_master == '[local]' and not portable_options.environment_type:
-        portable_options.environment_type == 'LOOPBACK'
+        portable_options.environment_type = 'LOOPBACK'
       return job_server.StopOnExitJobServer(FlinkJarJobServer(options))
     else:
       return flink_uber_jar_job_server.FlinkUberJarJobServer(flink_master)
 
+  @staticmethod
+  def add_http_scheme(flink_master):
+    flink_master = flink_master.strip()
+    # Add http protocol scheme if not provided
+    if not flink_master in ['[local]', '[auto]'] and \
+          not flink_master.startswith('http://'):
+      logging.info('Adding HTTP protocol scheme to flink_master parameter: '
+                   'http://%s', flink_master)
+      flink_master = 'http://' + flink_master
+    return flink_master
+
 
 class FlinkRunnerOptions(pipeline_options.PipelineOptions):
   @classmethod
   def _add_argparse_args(cls, parser):
     parser.add_argument('--flink_master',
-                        default='[auto]',
-                        help='Flink master address (host:port) to submit the'
-                             ' job against. Use "[local]" to start a local'
-                             ' cluster for the execution. Use "[auto]" if you'
-                             ' plan to either execute locally or submit through'
-                             ' Flink\'s CLI.')
+                        default='[local]',
```

Comment:
Per discussion the default should remain '[auto]'?

Reply:
The default was `[local]`. We had changed it to auto, but we don't know yet what that means in the context of `flink_runner.py`. So I'd suggest to leave it at `[local]` for now.

Reply:
Yes, this is the purpose of this PR and fixed here. Changing the default to `[auto]` broke it in the first place. Now it will work for `[auto]` or `[local]`. Still the more sensible default is probably `[local]` for Python at the moment. There is no real use for `[auto]` unless the job server is brought up with a config.

Reply:
On this note, we should add tests for `flink_runner.py`. Currently there are none: https://jira.apache.org/jira/browse/BEAM-8512

Reply:
The previous default of '[local]' that you restore here already had the above mentioned issue. This runner is not only used for the direct job submission but also for outputting generated jar files. We should verify that all the different submit mechanisms work.

Reply:
+1 I think these changes make sense. Although I would definitely like to see some tests (which I guess is [BEAM-8512](https://issues.apache.org/jira/browse/BEAM-8512))

Reply:
>+1 I think these changes make sense. Although I would definitely like to see some tests (which I guess is BEAM-8512)

Tests for the Python side are out of scope for this PR. There are currently none and creating them will require some effort. We will have to follow-up with this.

Reply:
Sounds good.

Reply:
@mxm yes, I believe that was the issue. I think the proper fix is to keep default at "auto" and change the behavior of the newly introduced Python REST API client to handle accordingly. Part of making changes here has to be verification. If there is no automation, then manual.

Reply:
@tweise I've fully verified that the current version of this PR works under all deployment scenarios, i.e. Python 2 / Python < 3.6 / Python >= 3.6, with `flink_master` set to `[local]`, `[auto]`, or an actual address.

Reply:
I retested the jar file upload with the current changes and it now works.

I'm OK with all changes in the PR except the default value modification.

A default value of '[local]' is inconsistent with that in the Java Flink runner and such inconsistencies have bitten us already elsewhere.

There is no reason for the change since '[auto]' already does the right thing, per verification, and accurately reflects the behavior: Run locally except when in Flink context.